src/Services/AvatarExistsLimitService.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+
+namespace WCM\User\Avatar\Services;
+
+
+class AvatarExistsLimitService implements ServiceInterface {
+
+	/** @var string */
+	private $key;
+
+
+	/**
+	 * @param string $key
+	 */
+	public function __construct( $key )
+	{
+		$this->key = $key;
+	}
+
+	/**
+	 * @param array $file
+	 * @return string
+	 */
+	public function setup( Array $file = [ ] )
+	{
+
+		$screen = filter_input( INPUT_POST, 'screen_id', FILTER_SANITIZE_STRING );
+		if ( ! in_array( $screen, [ 'profile', 'user-edit' ], TRUE ) )
+		{
+			return $file;
+		}
+
+		$user_id = filter_input( INPUT_POST, 'user_id', FILTER_SANITIZE_NUMBER_INT );
+
+		if( false === get_userdata($user_id) ){
+			$file['error'] = 'Cannot find user.';
+			return $file;
+		}
+
+		if( $this->userHasAvatar( $user_id ) ){
+			$file['error'] = 'This user already has an avatar set. To upload a new avatar delete the old one first.';
+			return $file;
+		}
+
+		return $file;
+	}
+
+	/**
+	 * @param int $user_id
+	 *
+	 * @return bool
+	 */
+	protected function userHasAvatar( $user_id ){
+
+		$avatar = get_user_meta( $user_id, $this->key, true );
```

Comment:
The `get_user_meta()` is a wrapper around [`get_metadata()`](https://developer.wordpress.org/reference/functions/get_metadata/), which returns `''` or `array()` – in case of the third argument set to `true`, it returns an empty string. So instead of using `!! $avatar`, we should just check

    return ! empty( get_user_meta( $user_id, $this->key, true ) )

as it's easier to read and checks against the real value instead of double negation of a true-ish value.